client/cli/src/commands/runcmd.rs

```diff
@@ -329,18 +330,18 @@ impl RunCmd {
 		let keyring = self.get_keyring();
 		let is_dev = self.shared_params.dev;
 		let is_light = self.light;
-		let is_authority = (self.validator || self.sentry || is_dev || keyring.is_some())
+		let is_authority = (self.validator || self.sentry.is_some() || is_dev || keyring.is_some())
 			&& !is_light;
 		let role =
 			if is_light {
-				sc_service::Roles::LIGHT
+				sc_service::Role::Light
```

Comment:
Sentry role never gets set here. This is because we're treating our role as `Authority` (since that's what we'll announce to the network), but we are using this `sentry_mode` flag to figure out that our local role is actually sentry (this is why I think we need to distinguish between network and local role).